hpx/parallel/executors/thread_execution.hpp

```diff
@@ -126,7 +126,8 @@ namespace hpx { namespace threads
     HPX_FORCEINLINE
     typename std::enable_if<
         hpx::traits::is_threads_executor<Executor>::value &&
-        std::is_same<Hint, hpx::threads::thread_schedule_hint>::value
+        std::is_same<typename hpx::util::decay<Hint>::type,
+            hpx::threads::thread_schedule_hint>::value
     >::type
     post(Executor && exec, F && f, Ts &&... ts, Hint && hint)
```

Comment:
How can there be an argument after a variadic one? I always assumed that the variadic list of arguments should be the last one.

Reply:
Sorry. Turns out I don't know what I'm doing after all.
Good job those variadic args are not actually used here. I've switched the order round now.
(gcc didn't mind about the variadic in the middle it seems)

Reply:
@biddisco may I suggest that you create a (completely separate) customization point for your executor that is enable_if'ed based on the guided_executor type? I think adding the additional condition on the scheduling hint may cause ambiguities.

Reply:
That's a good idea. Didn't think of that. I'll have a look later.